libraries/webInterface/gui/config.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @fileOverview
+ * This is a library that should be included in hardware interface's config.html pages in order
+ * to automatically add a websocket to the page and give them access to teh InterfaceConfig
+ * APIs, which will allow them to subscribe to realtime updates to the settings from the
+ * hardware interface's index.js
+ */
+
+let socketIoScript = {};
+let socketIoRequest = {};
+
+// Load socket.io.js synchronous so that it is available by the time the rest of the code is executed.
+function loadScriptSync(url, requestObject, scriptObject) {
+    requestObject = new XMLHttpRequest();
+    requestObject.open('GET', url, false);
```

Comment:
This may simplify the control flow of the page but it will destroy performance and potentially introduce bugs in Firefox and other browsers which have deprecated synchronous XHRs

It would be much safer to force the config.html pages to also include socket.io since they're already including this file or set up a less brittle asynchronous approach.

Reply:
Ah ok. I just copy-and-pasted from how the sockets for the logic block settings pages were set up. I should probably change those too to the async method.

Reply:
Oh, hmm. Yeah, that would be a nice project for the future since doing this sort of async thing adds a full roundtrip to the page load time. It's definitely not critical right now since everything is on local networks and very powerful servers

Reply:
I updated it to create pending stubs for the socket APIs and trigger them when the script asynchronously loads, as we do in object.js. When I get the chance I can make similar changes to the logic node settings